docs/gallery/plot_external_resources.py

```diff
@@ -196,20 +202,21 @@
 ###############################################################################
 # Using the add_ref method with a field
 # ------------------------------------------------------
-# In the above example, the ``field`` keyword argument was empty because the data
-# of the :py:class:`~hdmf.container.Data` object passed in for the ``container``
-# argument was being associated with a resource. However, you may want to associate
-# an attribute of a :py:class:`~hdmf.container.Data` object with a resource, or
-# a dataset or attribute of a :py:class:`~hdmf.container.Container` object with
-# a resource. To disambiguate between these different fields, you can set the
-# 'field' keyword.
+# It is important to keep in mind that the :py:class:`~hdmf.common.resources.Object` object
+# we add to the :py:class:`~hdmf.common.resources.ObjectTable` is the object
+# holding the attribute or the closest parent object. Recall that the objects in the
+# :py:class:`~hdmf.common.resources.ObjectTable` must have a 'data_type'.
+# A common example would be with the :py:class:`~hdmf.common.table.DynamicTable` class, which holds
+# :py:class:`~hdmf.common.table.VectorData` objects as columns. If we wanted to add an external
+# reference on a column from a :py:class:`~hdmf.common.table.DynamicTable`, then we would use the
+# column as the object and not the :py:class:`~hdmf.common.table.DynamicTable` (Refer to rule 10).
```

Comment:
This text implies that we should use``er.add_ref(container=genotypes.genotype_name, attribute=None)`` in the example below. I think it's fine to leave the example as is (since this is probably what most users will intuitively do), but we should mention here that ``add_ref`` internally resolves the object to the closest parent, so that `er.add_ref(container=genotypes, attribute='genotype_name')`` and ``er.add_ref(container=genotypes.genotype_name, attribute=None)`` will ultimatly both use the object_id of the ``genotypes.genotype_name`` VectorData column and not the object_id of the ``genotypes`` table.